packages/database/src/sync/buildEncounterLinkedSyncFilter.ts

```diff
@@ -1,18 +1,36 @@
 import { Utils } from 'sequelize';
+import { isObject, isString } from 'lodash';
+import { Model } from '../models/Model';
 
-export function buildEncounterLinkedSyncFilterJoins(tablesToTraverse: string[]) {
+export type JoinConfig =
+  | string
+  | {
+      model: typeof Model;
+      joinColumn: string;
+      required?: boolean;
+    };
+
+export function buildEncounterLinkedSyncFilterJoins(tablesToTraverse: JoinConfig[]) {
   return tablesToTraverse
     .slice(1)
-    .map(
-      (table, i) => `
-      LEFT JOIN ${table} ON ${tablesToTraverse[i]}.${Utils.singularize(table)}_id = ${table}.id
-    `,
-    )
+    .map((table, i) => {
+      const currentTable = isString(tablesToTraverse[i])
+        ? tablesToTraverse[i]
+        : tablesToTraverse[i]?.model.tableName;
+
+      const joinTable = isString(table) ? table : table.model.tableName;
+      const joinColumn = isString(table) ? `${Utils.singularize(table)}_id` : table.joinColumn;
+      const joinType = isObject(table) && table.required ? 'INNER' : 'LEFT';
+
+      return `
+        ${joinType} JOIN ${joinTable} ON ${currentTable}.${joinColumn} = ${joinTable}.id
+      `;
+    })
     .join('\n');
 }
 
 export function buildEncounterLinkedSyncFilter(
-  tablesToTraverse: string[], // e.g. [ 'survey_response_answers', 'survey_responses', 'encounters'] to traverse up from survey_response_answers
+  tablesToTraverse: JoinConfig[], // e.g. [ 'survey_response_answers', 'survey_responses', 'encounters'] to traverse up from survey_response_answers
```

Comment:
Could add an object to the e.g. in the comment

Reply:
Only thing is that the comment makes it more obvious that `encounters` needs to be included, though that could be done through a differently structured comment

Reply:
I am totally on board with the idea of letting the type do the talking in general though!